### DISCUSSION TITLE
Compiling implot demo program with MinGW

### DISCUSSION THREAD

**Discussion by cwlai2021** [epezent/implot#497] [Q&A] [answered]
Dear all,
I have been struggling for quite a while now. I saw some of the demos of implot and would like to try compiling implot to do some plotting. At this stage I simply want to try it so any demo (e.g., plot a sine function) will do. Nevertheless, most of the info I found was on how to compile implot's demo under MS visual studio. Since I am currently using c++ provided by MinGW I'd like to know how I can compile a demo of implot using MinGW. So far I have been successful in compiling imgui's example (main.cpp) using the Makefile in imgui-docking\examples\example_glfw_opengl2\Makefile, Alternatively, I can also compile the main.cpp in imgui-docking\examples\example_win32_directx11 on the command line:

export IMGUI_DIR=/c/tmp/imgui-docking
g++ -static -static-libgcc -static-libstdc++ ./main.cpp $IMGUI_DIR/backends/imgui_impl_dx11.cpp $IMGUI_DIR/backends/imgui_impl_win32.cpp $IMGUI_DIR/imgui*.cpp -I $IMGUI_DIR -I $IMGUI_DIR/backends   -ld3d11 -ld3dcompiler -lgdi32  -ldwmapi

For implot, however, I still can't figure how to compile a demo. More specifically, I can't find a demo main.cpp nor a Makefile. May I ask how I can, by using MinGW, compile a demo of implot (hopefully with a main.cpp and a Makefile), or using the command line similar to the above for the compilation? Your help is greatly appreciated. Thank you.

Kevin

**Comment by hinxx** [comment, marked as answer]
If you take a look at the implot Integration chapter in README.md you be on the right track to modify ImGui main.cpp with ImPlot function calls.

To see ImPlot demo window do as follows.

In imgui main.cpp, first include implot.h. Create ImPlot context after ImGui context with ImPlot::CreateContext(). Add a call to ImPlot::ShowDemoWindow() after the call to ImGui::ShowDemoWindow(). Also add ImPlot::DestroyContext() before ImGui::DestroyContext() to destroy the ImPlot context at exit.

Recompile and start the example..

**Reply by cwlai2021** [reply]
Thank you for your tips. I am trying to modify the main.cpp in one of imgui's examples. I can get it compiled, but it doesn't seem to run as expected. Will see if I can fix the problem. Thanks.

**Reply by hinxx** [reply]
Not sure what you mean that it does not run as expected..

There is a section called 'Extremely Important Note' which you might want to check out. Got bitten by it myself.

**Reply by cwlai2021** [reply]
My case is that the main.cpp I wrote compiled successfully into a .exe file. When run, however, the executable fails to even launch a window. So I must have written something wrong in my code. It'd be easier if a simple main.cpp and a Makefile can be provided by someone more experienced, but I understand it's my responsibility to write one myself.

**Reply by hinxx** [reply]
Show us what you got. Otherwise it is hard to tell what the issue is. I guess creating a github repo with all the code, compile commands and Makefile should not be that difficult.

**Reply by cwlai2021** [reply]
I deleted my files last week so I attempted to modify the main.cpp and Makefile again. Then I realized the reason why my executable failed to launch a window was because I made a silly mistake of not calling ImPlot::CreateContext(). The code runs as expected now.

I also attempted to plot a very simple graph with ImPlot::PlotLine(), but I only see the window without the line. But I guess I can fix this issue myself.

Just in case, I created a github repo for my trimmed down version at https://github.com/cwlai2021/my_imgui_implot.git
Thanks hinxx for the help.
